code/game/objects/effects/overlays.dm

```diff
@@ -20,37 +20,58 @@
 	spawn(10) qdel(src)
 
 /obj/effect/overlay/temp
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
 	anchored = 1
 	layer = 4.1
 	mouse_opacity = 0
 	var/duration = 10
 	var/randomdir = 1
+	var/loopless_icon = null //Workaround so that pooled effects that normally don't repeat will repeat when taken from the pool.
+
+/obj/effect/overlay/temp/Destroy()
+	..()
+	return QDEL_HINT_PUTINPOOL
 
 /obj/effect/overlay/temp/New()
 	if(randomdir)
 		dir = pick(cardinal)
+	if(loopless_icon)
+		flick("[loopless_icon]", src)

```

Comment:
This was the workaround.
I couldn't figure out a better way to make the icon repeat, as setting it to repeat indefinitely instead of once ALSO causes odd icon behavior with pooling, such as starting in the middle of the animation.